app/src/main/java/protect/card_locker/LoyaltyCardCursorAdapter.java

```diff
@@ -355,6 +358,14 @@ public void setIconBackgroundColor(int color) {
             mIconBackgroundColor = color;
             mCardIcon.setBackgroundColor(color);
         }
+
+        public void toogleArchived(boolean enable){
+            if (enable) {
+                mArchived.setVisibility(View.VISIBLE);
+            } else {
+                mArchived.setVisibility(View.GONE);
+            }
+        }
```

Comment:
The way i think about this is that a archived card can also be a starred card, this way if you unarchive it it would show that is starred. If you put a enum like this:

```java 
public enum LoyaltyCardStateIcon{
Archived,
Starred,
None}
```
A card could not be archived and starred  at the same time

Reply:
Makes sense, then turning `toggleStar` into `showCardIcons(boolean starred, boolean archived, boolean colorByTheme)` would probably be better. Because probably the archive icon will also need the same foreground and background handling as the star icon.